client/my-sites/post-type-filter/author-segmented.jsx

```diff
@@ -0,0 +1,63 @@
+/**
+ * External Dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { compact, map } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Gravatar from 'components/gravatar';
+import NavItem from 'components/section-nav/item';
+import NavSegmented from 'components/section-nav/segmented';
+import { getCurrentUser } from 'state/current-user/selectors';
+import { getSiteSlug } from 'state/sites/selectors';
+
+const AuthorSegmented = ( { author, siteSlug, statusSlug, translate, user } ) => {
+	const scopes = {
+		me: translate( 'Me', { context: 'Filter label for posts list' } ),
+		everyone: translate( 'Everyone', { context: 'Filter label for posts list' } )
+	};
+
+	return (
+		<NavSegmented label={ translate( 'Author', { context: 'Filter group label for segmented' } ) }>
+			{ map( scopes, ( label, scope ) => {
+				const isMe = 'me' === scope;
+				const path = compact( [
+					isMe ? '/posts/my' : '/posts',
+					statusSlug,
+					siteSlug
+				] ).join( '/' );
+
+				return (
+					<NavItem key={ scope }
+						path={ path }
+						selected={ isMe === !! author } >
+							{ label }
+							{ isMe && <Gravatar size={ 16 } user={ user } /> }
+					</NavItem>
+					);
+			} ) }
+		</NavSegmented>
+	);
+};
+
+AuthorSegmented.propTypes = {
+	author: PropTypes.number,
+	siteId: PropTypes.number,
+	statusSlug: PropTypes.string,
+	// Connected Props
+	siteSlug: PropTypes.string,
+	translate: PropTypes.func.isRequired,
```

Comment:
Nit: Translate isn't a connected prop.

Reply:
Right, it's not coming from `connect()`. I've started to list other props coming from HOCs under this header too, tho. Maybe I should rename these comments to `// From HOCs`... :thinking: